odoo/http.py

```diff
@@ -1055,7 +1055,9 @@ def check_security(self):
         """
         if not self.db or not self.uid:
             raise SessionExpiredException("Session expired")
-        security.check(self.db, self.uid, self.password)
+        # here we check if the session is still valid
+        if not security.check_session(self.db, self.sid, self.uid, self.session_token):
```

Comment:
Maybe just pass the session object and check_session can check whatever it wants? Same for compute_session_token?